### PR TITLE
Generate a quote

### PR DESCRIPTION
**Pull request by teneightfive**
This adds the controller and middleware logic to handle generating
a quote from an order.

Has some basic error handling on API responses but more detailed
user errors will come later.

### CODE REVIEW

File: src/apps/omis/apps/view/controllers.js [hunk 1]
```diff
@@ -8,12 +8,18 @@ function renderWorkOrder (req, res) {
...
-    .breadcrumb('Work order')
     .render('omis/apps/view/views/work-order', {
       values,
     })
 }
 
+function renderQuote (req, res) {
+  res
+    .breadcrumb('Quote')
+    .render('omis/apps/view/views/quote')
```
Comment (feedmypixel) [inline src/apps/omis/apps/view/controllers.js]: can the naming for this folder structure `/view/views` be improved?

File: src/apps/omis/apps/view/middleware.js [hunk 2]
```diff
@@ -1,11 +1,59 @@
...
+      return next()
+    }
+
+    next(error)
+  }
+}
+
+async function generateQuote (req, res, next) {
+  try {
+    await Order.createQuote(req.session.token, res.locals.order.id)
```
Comment (feedmypixel) [inline src/apps/omis/apps/view/middleware.js]: `res.locals.order.id` could be put into a local const as its used twice

File: src/apps/omis/apps/view/views/_layouts/view.njk [hunk 3]
```diff
@@ -18,9 +15,8 @@
 
     {{ MetaList({
       items: [
-        { "label": "Status", "value": "Draft" },
-        { "label": "Created on", "value": order.created_on | formatDate },
-        { "label": "Created by", "value": order.created_by }
+        { "label": "Created on", "value": order.created_on | formatDateTime },
```
Comment (feedmypixel) [inline src/apps/omis/apps/view/views/_layouts/view.njk]: I presumed the quotes in nunjucks for non html work were single? (html is double) I may be wrong

Comment (teneightfive) [reply src/apps/omis/apps/view/views/_layouts/view.njk]: I'm not sure to be honest. I probably haven't been very good at using a standard set within nunjucks files. Will try and use this pattern moving forward.

File: assets/stylesheets/_deprecated/components/_button.scss [hunk 4]
```diff
@@ -10,6 +10,10 @@
   }
 }
 
+.button--destructive {
+  @include button ($red);
```
[the remark this answers is not in the document]
Comment (teneightfive) [reply assets/stylesheets/_deprecated/components/_button.scss]: This was a hangover from the branch is was rebased on top of (#525). This change was fixed there and isn't present here any more :)

File: src/apps/omis/apps/view/controllers.js (hunk 1 above)
Comment (teneightfive) [reply src/apps/omis/apps/view/controllers.js]: It's not the best but `view` seemed to best describe the sub-app state. Happy for any alternative suggestions for the _view_ app.

Definitely outside of scope for this change though.

Comment (feedmypixel) [reply src/apps/omis/apps/view/controllers.js]: > Definitely outside of scope for this change though.

agreed - something a little more semantic may help explain what it is (as discussed)

**Pull request merged**